### PR TITLE
Add Python 3.6, drop Python 3.3

### LINKED ISSUE #3: Get unit test coverage to 100%
And pin it there via `setup.cfg`.

### PR DESCRIPTION
**Pull request by jamadden**
- Badges and project gardening.
- Whitespace cleanup (it was hard to see actual issues through all the whitespace warnings that pep8/pylint gave out).
- Coverage environment and coveralls
- 100 % coverage (fixes #3, though I'm not too happy about all the "pragma: no cover" in stdom.py for all the unused methods)
- Some docbook fixes and additions discovered in the process of getting coverage up

This turned out to be a big PR. I've left all the commits unsquashed for now in the hopes of making it easier to review.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+=========
+ Changes
+=========
 
 4.2.0 (unreleased)
-------------------
+==================
 
-- Add support for Python 3.5.
+- Add support for Python 3.5 and 3.6.
```
Comment (icemac) [inline CHANGES.rst:8]: `and PyPy2 + PyPy3` as they where not tested previously.

File: src/zope/structuredtext/docbook.py [hunk 2]
```diff
@@ -282,7 +282,9 @@ def read(self):
                '<book>\n')
         out = out + '<title>%s</title>\n' % self.title
         for chapter in self.chapters:
-            out = out + chapter + '\n</book>\n'
+            out = out + chapter + '\n'
+
+        out += '\n</book>\n'
```
Comment (icemac) [inline src/zope/structuredtext/docbook.py:287]: Should this be mentioned in the changelog as bugfix?

**Review by icemac — APPROVED** [review]
Thank you for your hard work to reach complete coverage and fixing bugs.

LGTM but I mostly scanned the individual commits for something obvious and did not find something.

File: src/zope/structuredtext/docbook.py (hunk 2 above)
Comment (jamadden) [reply src/zope/structuredtext/docbook.py:287]: Yup, thanks.

File: CHANGES.rst (hunk 1 above)
Comment (jamadden) [reply CHANGES.rst:8]: Good point. Thanks!

**Pull request merged**